python/pfs/utils/optimizeBlackDots.py

```diff
@@ -0,0 +1,501 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from scipy.optimize import minimize
+from lsst.geom import AffineTransform
+from pfs.drp.stella.math import evaluateAffineTransform
+
+
+class OptimizeBlackDots():
+    """Optimization of position of black dots
+
+    Class with functions for modifying the positions of black dots
+    for PFS spectrograph
+    """
+
+    def __init__(self, dots, list_of_mcs_data_all, list_of_descriptions):
+        """
+        Parameters
```

Comment:
numpydoc for an `__init__` goes in the class docstring (because you call the class to instantiate).